sdks/java/io/synthetic/src/test/java/org/apache/beam/sdk/io/synthetic/CombineLoadIT.java

```diff
@@ -0,0 +1,102 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.synthetic;
+
+import static org.apache.beam.sdk.io.synthetic.SyntheticSourceTestUtils.fromString;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import org.apache.beam.sdk.options.Default;
+import org.apache.beam.sdk.options.Description;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.options.PipelineOptionsValidator;
+import org.apache.beam.sdk.options.Validation;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.testing.TestPipelineOptions;
+import org.apache.beam.sdk.transforms.Combine;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.Mean;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Performance fanout test for {@link Combine}. */
+@RunWith(JUnit4.class)
+public class CombineLoadIT {
+
+  private static Options options;
+
+  private static SyntheticBoundedIO.SyntheticSourceOptions syntheticSourceOptions;
+
+  @Rule public TestPipeline pipeline = TestPipeline.create();
+
+  /** Pipeline options for the test. */
+  public interface Options extends TestPipelineOptions {
+
+    @Description("The JSON representation of SyntheticBoundedInput.SourceOptions.")
+    @Validation.Required
+    String getInputOptions();
+
+    void setInputOptions(String inputOptions);
+
+    @Description("The number of gbk to perform")
+    @Default.Integer(5)
+    Integer getFanout();
+
+    void setFanout(Integer shuffleFanout);
+  }
+
+  @BeforeClass
+  public static void setup() throws IOException {
+    PipelineOptionsFactory.register(Options.class);
+
+    options =
+        PipelineOptionsValidator.validate(
+            Options.class, TestPipeline.testingPipelineOptions().as(Options.class));
+
+    syntheticSourceOptions = fromString(options.getInputOptions());
+  }
+
+  @Test
+  public void combineLoadTest() {
+    PCollection<KV<byte[], byte[]>> input =
+        pipeline.apply(SyntheticBoundedIO.readFrom(syntheticSourceOptions));
+
+    PCollection<Integer> numbers =
+        input.apply("Get numbers from bytes", ParDo.of(new ByteToIntFn()));
+
+    for (int branch = 0; branch < options.getFanout(); branch++) {
+      numbers.apply(String.format("Combine (%s)", branch), Mean.globally());
```

Comment:
We'll want to combine per key here.

Reply:
@pabloem see this PR: https://github.com/apache/beam/pull/6985 for current implementation